kube-runtime/src/watcher.rs

```diff
@@ -23,6 +24,8 @@ pub enum Error {
     WatchFailed(#[source] kube_client::Error),
     #[error("too many objects matched search criteria")]
     TooManyObjects,
+    #[error("retriable watch error")]
+    BackoffRetriable,
```

Comment:
This is an internal implementation detail that we shouldn't expose.

Reply:
i guess if we make a wrapper for `watcher` instead inside util then we can keep the error constrained to that

Reply:
have hidden the error entirely and made the backoff a stream wrapper fn